routes/user_r.js

```diff
@@ -1,6 +1,11 @@
 const express = require('express');
 const router = new express.Router();
 const usersController = require('../controllers/users_c.js');
+const validate = require('../utilities/validation.js');
+
+router.get('/', usersController.getAllUsers);
+
+router.get('/:username', usersController.getUserByUsername);
```

Comment:
Please refer to my comment in [this Teams conversation](https://teams.microsoft.com/l/message/19:4887936592964ee48d090a0e1414efb6@thread.tacv2/1732894484040?tenantId=e6ac1d1f-d695-4ef1-91d4-94cddef8be11&groupId=255572aa-d6a5-4e62-af82-991fafb918c3&parentMessageId=1732821306845&teamName=Fall%202024%20CSE%20341%20Web%20Services%20Block%202%20Section%205&channelName=Team%2006&createdTime=1732894484040) about using the username vs. an id and the reasoning behind it.